src/btree/bt_read.c

```diff
@@ -68,6 +68,129 @@ __row_instantiate(WT_SESSION_IMPL *session,
 	return (0);
 }
 
+/*
+ * __wt_find_lookaside_upd --
+ * 	Scan the lookaside for a record the btree cursor wants to position on.
+ * 	Create an update for the record and return to the caller.
+ */
+int
+__wt_find_lookaside_upd(WT_SESSION_IMPL *session,
+    WT_CURSOR_BTREE *cbt, WT_UPDATE **updp, bool ignore_birthmark)
+{
+	WT_CACHE *cache;
+	WT_CURSOR *cursor;
+	WT_DECL_RET;
+	WT_ITEM *key;
+	WT_ITEM las_key, las_value;
+	WT_REF *ref;
+	WT_UPDATE *upd;
+	wt_timestamp_t durable_timestamp, las_timestamp;
+	size_t incr;
+	uint64_t las_counter, las_pageid, las_txnid;
+	uint32_t las_id, session_flags;
+	uint8_t prepare_state, upd_type;
+	int cmp;
+	bool locked, upd_visible;
+
+	cache = S2C(session)->cache;
+	cursor = NULL;
+	session_flags = 0;		/* [-Werror=maybe-uninitialized] */
+	ref = cbt->ref;
+	las_pageid = ref->page_las->las_pageid;
+	key = &cbt->iface.key;
+	*updp = upd = NULL;
+
+	/* Open a lookaside table cursor. */
+	__wt_las_cursor(session, &cursor, &session_flags);
+
+	/*
+	 * The lookaside records are in key and update order, that is, there
+	 * will be a set of in-order updates for a key, then another set of
+	 * in-order updates for a subsequent key. We scan through all of the
+	 * updates till we locate the one we want.
+	 */
+	WT_PUBLISH(cache->las_reader, true);
+	__wt_readlock(session, &cache->las_sweepwalk_lock);
+	WT_PUBLISH(cache->las_reader, false);
+	locked = true;
+
+	for (ret = __wt_las_cursor_position(cursor, las_pageid);
+	    ret == 0;
+	    ret = cursor->next(cursor)) {
+		WT_ERR(cursor->get_key(cursor,
+		    &las_pageid, &las_id, &las_counter, &las_key));
+
+		/* Stop before crossing over to the next page block */
+		if (las_pageid != ref->page_las->las_pageid)
+			break;
+
+		/*
+		 * Keys are sorted in an order, skip the ones before the desired
+		 * key, and bail out if we have crossed over the desired key and
+		 * not found the record we are looking for.
+		 */
+		WT_ERR(__wt_compare(session,
+		    S2BT(session)->collator, &las_key, key, &cmp));
+		if (cmp < 0)
+			continue;
+		else if (cmp > 0)
+			break;
+
+		WT_ERR(cursor->get_value(
+		    cursor, &las_txnid, &las_timestamp,
+		    &durable_timestamp, &prepare_state, &upd_type, &las_value));
+
+		/*
+		 * It is safe to assume that the updates are sorted newest to
+		 * the oldest. We can quit searching after finding the newest
+		 * visible record.
+		 */
+		upd_visible =
+		    __wt_txn_visible(session, las_txnid, las_timestamp);
+		if (!upd_visible)
+			continue;
```

Comment:
Same here. If `ret` is 0 here then we're going to break out of the loop which sounds like not what you're intending.